components/omega/src/ocn/OceanRun.cpp

```diff
@@ -0,0 +1,63 @@
+//===-- ocn/OceanRun.cpp - Run Ocean Model ----------------------*- C++ -*-===//
+//
+// The ocnRun method advances the model forward from CurrTime until the
+// EndAlarm rings.
+//
+//===----------------------------------------------------------------------===//
+
+#include "OceanDriver.h"
+#include "OceanState.h"
+#include "TimeStepper.h"
+
+namespace OMEGA {
+
+int ocnRun(TimeInstant &CurrTime, ///< [inout] current sim time
+           Alarm &EndAlarm        ///< [inout] alarm to end simulation
+) {
+
+   // error code
+   I4 Err = 0;
+
+   // fetch default OceanState and TimeStepper
+   OceanState *DefOceanState   = OceanState::getDefault();
+   TimeStepper *DefTimeStepper = TimeStepper::getDefault();
+
+   TimeInterval TimeStep, ZeroInterval;
+
+   // set simulation clock and attach EndAlarm
+   TimeStep = DefTimeStepper->getTimeStep();
+   Clock OmegaClock(CurrTime, TimeStep);
+   Err = OmegaClock.attachAlarm(&EndAlarm);
+
+   if (TimeStep == ZeroInterval) {
+      LOG_ERROR("ocnRun: TimeStep must be initialized");
+      ++Err;
+   }
+
+   I8 IStep = 0;
+
+   // time loop, integrate until EndAlarm or error encountered
+   while (Err == 0 && !(EndAlarm.isRinging())) {
+
+      // advance clock
+      OmegaClock.advance();
+      ++IStep;
+
+      // call forcing routines, anything needed pre-timestep
+
+      // do forward time step
+      TimeInstant SimTime = OmegaClock.getPreviousTime();
```

Comment:
Is there an efficient method to obtain the elapsed time since time zero from TimeInstant `SimTime`?

As the manufactured solution requires the elapsed time in seconds (in R8) to generate tendencies at a given time, a straightforward procedure to obtain the elapsed time since time zero from TimeInstant may be necessary. In my test for this PR with Polaris, I managed it in a somewhat tricky way.

I'm afraid that my question is not directly related to this standalone driver PR, but it would be helpful if we could verify this in advance

Reply:
No problem. The Clock object stores the initial time and can be retrieved with `OmegaClock.getStartTime()`, which returns a TimeInstant, and you can take the difference of two TimeInstants to get a TimeInterval, i.e. `TimeInterval ElapsedTime = StartTime - SimTime`.

Reply:
That's actually what I did. And if I want to get the `ElapsedTime` in the R8 type, is there an efficient way to do it?

What I did was using 1) `Calendar::get` to get YYYY, MM, DD, HH, etc, 2) `Calendar::getElapsedTime` and then 3) `TimeFrac::getSeconds`. Maybe I missed something, and I believe there is a more efficient way to do it. 

One thing I found that it is more convenient if I can retrieve to the `ElapsedTime` in `TimeInstant` by creating an additional function returning the `ElapsedTime` in `TimeInstant`. For example,
`TimeFrac getElapsedTime(void) const;` in [TimeMgr.cpp](https://github.com/hyungyukang/Omega/blob/9d53943c81ac4f70a0ba6033adc1cddc2b4fb8c4/components/omega/src/infra/TimeMgr.cpp#L4324)
`TimeFrac TimeInstant::getElapsedTime(void) const { return ElapsedTime; }` in [TimeMgr.h](https://github.com/hyungyukang/Omega/blob/9d53943c81ac4f70a0ba6033adc1cddc2b4fb8c4/components/omega/src/infra/TimeMgr.h#L601)


And then I used the codes below to obtain the `ElapsedTime` in the R8 type:
```
TimeFrac ElapsedTime = Time.getElapsedTime();  ////< TimeInstant Time
R8 ElapsedTimeSec = ElapsedTime.getSeconds();
```

Reply:
There is an existing `get` function for TimeInterval that is overloaded to different data types and requires an enum to declare the units you want the interval in. So the elapsed time in seconds as an R8 can be retrieved via:
```
TimeInterval ElapsedTime = SimTime - StartTime;
R8 ElapsedTimeSec;
I4 Err = ElapsedTime.get(ElapsedTimeSed, TimeUnits::Seconds);
```

Reply:
That's exactly what I wanted to do. I just tested it, and it worked well. Thanks a lot!